### PR TITLE
chore: Quality pass on the bundling chapter

### PR DESCRIPTION
**Pull request by bebraw**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -2,6 +2,9 @@
   "env": {
     "es6": true
   },
+  "parserOptions": {
+    "sourceType": "module"
```
Comment (sapegin) [inline .eslintrc]: Why do we need it? Looks like all examples in the book are actually `script`s.

File: manuscript/packaging/03-bundling.md [hunk 2]
```diff
@@ -1,6 +1,14 @@
 # Bundling Packages
 
-If you author your packages using a source format that works directly with Node, you can avoid additional processing. npm `engines` field allows you to control the minimum version of Node to support and it accepts a range. Consider the example adapted from [npm documentation](https://docs.npmjs.com/files/package.json#engines):
+If you author your packages using a source format that works directly with Node, you can avoid additional processing. In this case it's enough to point *package.json* `main` field to the source as discussed in the previous chapter.
```
Comment (sapegin) [inline manuscript/packaging/03-bundling.md]: This section is too abstract. You’re discussing several things here:

* what browsers / Node versions we are supporting;
* what to do if we want to use language feature that aren’t supported by our targets, or another language;
* how to support modern bundlers (tree shaking).

File: manuscript/packaging/03-bundling.md [hunk 3]
```diff
@@ -195,23 +201,20 @@ T> [Webpack and Rollup: the same but different](https://medium.com/webpack/webpa
 
 T> You can also use webpack and Rollup together through [rollup-loader](https://www.npmjs.com/package/rollup-loader) and leave JavaScript handling to Rollup as this enables tree shaking through Rollup and allows you to use Rollup specific plugins.
 
-## Scope Hoisting
-
-TODO: rollup, webpack
-TODO: what's the point
+T> Both webpack and Rollup support *scope hoisting*. It's a performance oriented feature that pulls modules into a single scope instead of writing separate scope for each module. This reduces the overhead related to function calls.
```
Comment (sapegin) [inline manuscript/packaging/03-bundling.md]: Maybe something like “wrapping each module into its own function” to make clearer what “scope” means here?

File: .eslintrc (hunk 1 above)
Comment (bebraw) [reply .eslintrc]: There's

```javascript
function demo() {
  console.log('demo');
}

export default demo;
```

which blew up without that. I had disabled the example before but then realized tuning ESLint is far better.

File: manuscript/packaging/03-bundling.md (hunk 2 above)
Comment (bebraw) [reply manuscript/packaging/03-bundling.md]: Good point. I'll expand and rethink this a bit.

File: manuscript/packaging/03-bundling.md (hunk 3 above)
Comment (bebraw) [reply manuscript/packaging/03-bundling.md]: Yeah. I guess ideally we would have some code here to show the actual transformation.

File: .eslintrc (hunk 1 above)
Comment (sapegin) [reply .eslintrc]: OK, if it doesn’t break anything else ;-)

**Comment by bebraw** [comment]
I restructured the chapters a bit based on the feedback. The new structure feels better to me.

File: manuscript/packaging/05-standalone-builds.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# Standalone Builds
+
+The scenarios covered in the previous chapter are enough if you consume packages through npm. That is not the case always, though, as there may be users that prefer a *standalone build*. Generating one is an optional step and you may skip this entirely given consuming packages through npm is the standard.
```
Comment (sapegin) [inline manuscript/packaging/05-standalone-builds.md]: I think it’s more about using a bundler than using npm.

Comment (bebraw) [reply manuscript/packaging/05-standalone-builds.md]: Yeah, I'll amend that. Thanks.

**Pull request merged**